SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp

```diff
@@ -735,13 +735,13 @@ void SSpatialGDKSimulatedPlayerDeployment::OnPrimaryDeploymentNameCommited(const
 void SSpatialGDKSimulatedPlayerDeployment::OnCheckedUsePinnedVersion(ECheckBoxState NewCheckedState)
 {
 	USpatialGDKEditorSettings* SpatialGDKSettings = GetMutableDefault<USpatialGDKEditorSettings>();
-	SpatialGDKSettings->SetUseGDKPinnedRuntimeVersion(NewCheckedState == ECheckBoxState::Checked);
+	SpatialGDKSettings->SetUseGDKPinnedRuntimeVersion(SpatialGDKSettings->RuntimeVariant, NewCheckedState == ECheckBoxState::Checked);
```

Comment:
A slight concern here:
If you have previously specified the non-pinned version for local runtime, but are now using the pinned one, and then uncheck 'Use GDK Pinned Version' in the Cloud Deployment window (e.g. to run with a specific runtime in the cloud), it will start using the non-pinned version for local runtime as well, which would probably be unexpected.
How about having separate 'Use pinned version' values for cloud and local?

Reply:
Having to specify both version if you untick it in the settings sounds reasonable to me.
But here, this is one of the mildly annoying thing about the cloud deployment window, where everything is reflected to the settings. Making the "Use pinned version" checkbox modification transient in the cloud deployment window sounds reasonable. Can you create a ticket to do it ?

Reply:
I don't think it should be transient. Everything you specify in the cloud deployment window is saved so the user never has to re-enter details.

Reply:
I added a setting for both local and cloud now to address this.